src/products/index.ts

```diff
@@ -17,26 +18,46 @@ export type Transformer = (product: Product) => { data: object, variants: object
 @alias('products')
 @tag('gb-products', require('./index.html'))
 class Products {
-
   structure: Structure = this.config.structure;
+
   state: Products.State = {
     products: []
   };
+
+  init() {
+    switch (this.props.storeSection) {
+      case StoreSections.PAST_PURCHASES:
+        this.flux.on(Events.PAST_PURCHASE_PRODUCTS_UPDATED, this.updateProducts);
+        break;
+      case StoreSections.SEARCH:
+        this.flux.on(Events.PRODUCTS_UPDATED, this.updateProducts);
+      break;
+    }
+  }
+
   productTransformer: Transformer = ({ data, meta }: Product) =>
     ({ ...ProductTransformer.transformer(this.structure)(data), meta })
 
-  init() {
-    this.flux.on(Events.PRODUCTS_UPDATED, this.updateProducts);
+  updateProducts = (products: any = []) => {
+    switch (this.props.storeSection) {
```

Comment:
why not make separate update functions for different section of the store? Switch cases are not saving any code here